### DISCUSSION TITLE
Un-injest a document

### DISCUSSION THREAD

**Discussion by Oshirowanen** [imartinez/privateGPT#996] [Q&A]
How would I un-injest an injected document?

**Comment by patchworkfish** [comment]
Great project @imartinez. Thank you!
Landed here asking the same question and did a little dig, @Oshirowanen.

The injest process creates a db located at `privateGPT/db/chroma.sqlite3`.
The `db` folder also contains a folder with a GUID name containing 4 binary files, notably `data_level0.bin`.

Deleting all contents of `privateGPT/db/` allows you to start fresh and injest a new set of documents.

The source data is stored in `privateGPT/db/chroma.sqlite3` and can be queried and the relevant data for each source can be identified, potentially allowing a script to remove data from here. If the appropriate data is removed from this DB, is it removed from inference? I'll assume not.

Perhaps the contents of `privateGPT/db/{GUID}/` could be regenerated from the data held within `privateGPT/db/chroma.sqlite3` after modification without re-loading files?

Any guidance greatly appreciated!
